hpx/util/detail/pack_traversal_impl.hpp

```diff
@@ -142,9 +143,69 @@ namespace util {
                 return Base<NewType, decltype(allocator)>(std::move(allocator));
             }
 
+            /// Returns the default iterators of the container in case
+            /// the container was passed as an l-value reference.
+            /// Otherwise move iterators of the container are returned.
+            template <typename C, typename = void>
+            class container_accessor
+            {
+                C const& container_;
+
+            public:
+                container_accessor(C const& container)
+                  : container_(container)
+                {
+                }
+
+                auto begin() -> decltype(container_.begin())
+                {
+                    return container_.begin();
+                }
+
+                auto end() -> decltype(container_.end())
+                {
+                    return container_.end();
+                }
+            };
+            template <typename C>
+            class container_accessor<C,
+                typename std::enable_if<
+                    std::is_rvalue_reference<C&&>::value>::type>
+            {
+                C&& container_;
+
+            public:
+                container_accessor(C&& container)
+                  : container_(std::move(container))
+                {
+                }
+
+                auto begin()
```

Comment:
The `begin` and `end` method can't be a const references because we must allow the mapping from non const l-value references too (this was fixed today).